### PR TITLE
[SYCL]Replaced LoadLibraryA() with LoadLibraryEx()

### PR DESCRIPTION
**Pull request by eunakim0103**
Replaced LoadLibraryA() with LoadLibraryEx() and added SetDllDirectoryA() to remove the working directory from the standard search path. LoadLibraryExA() was used because "path" is of type c-string.

- LoadLibraryExA() <-- for ANSI (ASCII)
- LoadLibraryExW() <-- for Unicode

The libloaderapi.h header defines LoadLibraryEx as an alias which automatically selects the ANSI or Unicode version of this function based on the definition of the UNICODE preprocessor constant.
https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexw

HMODULE LoadLibraryEx(
[in] LPCSTR lpLibFileName, //==> library path as a string
HANDLE hFile, //==> reserved as NULL for future use
[in] DWORD dwFlags //==> dwFlags determine the behavior of LoadLibaryEx(). When it is NULL, it behaves same as LoadLibaryA()
);
https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa

### CODE REVIEW

File: sycl/source/detail/windows_pi.cpp [hunk 1]
```diff
@@ -31,7 +31,8 @@ void *loadOsLibrary(const std::string &LibraryPath) {
   if (!SetDllDirectoryA("")) {
     assert(false && "Failed to update DLL search path");
   }
-  auto Result = (void *)LoadLibraryA(LibraryPath.c_str());
+  // auto Result = (void *)LoadLibraryA(LibraryPath.c_str());
```
Comment (eunakim0103) [inline sycl/source/detail/windows_pi.cpp:34]: Will remove this comment

File: sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp [hunk 2]
```diff
@@ -151,25 +151,26 @@ void preloadLibraries() {
   MapT &dllMap = getDllMap();
 
   std::string ocl_path = LibSYCLDir + __SYCL_OPENCL_PLUGIN_NAME;
-  dllMap.emplace(ocl_path, LoadLibraryA(ocl_path.c_str()));
+  dllMap.emplace(ocl_path, LoadLibraryExA(ocl_path.c_str(), NULL, NULL));
```
Comment (againull) [inline sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp:154]: I don't mind this change, but it seems like if flags are NULL then LoadLibraryExA is doing the same as LoadLibraryA.
See https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa

> If no flags are specified, the behavior of this function is identical to that of the [LoadLibrary](https://learn.microsoft.com/en-us/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya) function. This parameter can be one of the following values.

**Review by againull — APPROVED** [review]

**Comment by eunakim0103** [comment]
My fix passed all tests before and I removed one comment afterwards. The test fail in SYCL E2E on AWS CUDA is not relevant to my code. This PR is ready to merge @intel/llvm-gatekeepers.

**Comment by dm-vodopyanov** [comment]
@eunakim0103 post-merge nit comment: avoid using `NULL` in C++ code, instead use `nullptr`.

**Pull request merged**